distributed/shuffle/_rechunk.py

```diff
@@ -156,32 +155,25 @@ class ShardID(NamedTuple):
     shard_index: NIndex
 
 
-def rechunk_slicing(
-    old: ChunkedAxes, new: ChunkedAxes
-) -> dict[NIndex, list[tuple[ShardID, NSlice]]]:
-    """Calculate how to slice the old chunks to create the new chunks
+SplitChunk: TypeAlias = list[tuple[int, int, slice]]
+SplitAxis: TypeAlias = list[SplitChunk]
+SplitAxes: TypeAlias = list[SplitAxis]
 
-    Returns
-    -------
-        Mapping of each old chunk to a list of tuples defining where each slice
-        of the old chunk belongs. Each tuple consists of the index
-        of the new chunk, the index of the slice within the composition of slices
-        creating the new chunk, and the slice to be applied to the old chunk.
-    """
-    from dask.array.rechunk import intersect_chunks
-
-    ndim = len(old)
-    intersections = intersect_chunks(old, new)
```

Comment:
OK, so previously we would make the full cartesian product of intersections up front.

Reply:
Yes, previously we calculated the mapping information from each individual input chunk to each individual output chunk. This is results in the cartesian product of the chunks on the chunked axes.